cmd/kubeadm/app/cmd/util/join_test.go

```diff
@@ -0,0 +1,240 @@
+		Server:                   "test-server:6443",
+		ClusterName:              "somecluster",
+		ContextName:              "test@somecluster",
+		User:                     "test",
+	})
+}
+
+func generateMissingCAKubeConfig() *clientcmdapi.Config {
+	return generateKubeConfig(kubeConfigSpec{
+		Server:      "test-server:6443",
+		ClusterName: "somecluster",
+		ContextName: "test@somecluster",
+		User:        "test",
+	})
+}
+
+func generateInvalidFilePathKubeConfig() *clientcmdapi.Config {
+	return generateKubeConfig(kubeConfigSpec{
+		CertificateAuthorityFile: "invalid-file",
+		Server:                   "test-server:6443",
+		ClusterName:              "somecluster",
+		ContextName:              "test@somecluster",
+		User:                     "test",
+	})
+}
+
+func generateInvalidCAKubeConfig() *clientcmdapi.Config {
+	return generateKubeConfig(kubeConfigSpec{
+		CertificateAuthorityData: "invalid-data",
+		Server:                   "test-server:6443",
+		ClusterName:              "somecluster",
+		ContextName:              "test@somecluster",
+		User:                     "test",
+	})
+}
+
+func TestGetJoinWorkerCommand(t *testing.T) {
+	tests := []struct {
+		name         string
+		kubeConfig   *clientcmdapi.Config
+		expectError  bool
+		errorMessage string
+		token        string
+	}{
+		{
+			name:        "Success with valid kubeconfig and token",
+			kubeConfig:  generateValidKubeConfig(),
+			token:       "test-token",
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tmpDir, err := os.MkdirTemp("", "kubeadm-images-test")
+			if err != nil {
+				t.Fatalf("Unable to create temporary directory: %v", err)
+			}
+			defer func() {
+				err := os.RemoveAll(tmpDir)
+				if err != nil {
+					t.Fatalf("Unable to remove temporary directory: %v", err)
+				}
+			}()
+
+			configFilePath := filepath.Join(tmpDir, "test-config-file")
+			err = writeKubeConfigToFile(tt.kubeConfig, configFilePath)
+			require.NoError(t, err)
+
+			_, err = GetJoinWorkerCommand(configFilePath, tt.token, false)
```

Comment:
for err != nil should we verify the output string to be the expected one?

Reply:
Yes, fixed, thanks